src/XMakeTasks/AssemblyDependency/AssemblyInformation.cs

```diff
@@ -592,4 +599,214 @@ private void FreeAsmMeta(IntPtr asmMetaPtr)
+                    // the base file offset of the section.
+
+                    var sectionOffset = optionalHeaderOffset + optionalPEHeaderSize;
+
+                    // Read all section headers, we need them to make RVA to
+                    // offset conversions.
+
+                    var sections = new HeaderInfo [numberOfSections];
+                    for (int n = 0; n < numberOfSections; n++)
+                    {
+                        // At offset 8 of the section is the section size
+                        // and base RVA. At offset 20 there is the file offset
+                        sr.BaseStream.Position = sectionOffset + 8;
+                        var sectionSize = sr.ReadUInt32();
+                        var sectionRva = sr.ReadUInt32();
+                        sr.BaseStream.Position = sectionOffset + 20;
+                        var sectionDataOffset = sr.ReadUInt32();
+                        sections[n] = new HeaderInfo {
+                            VirtualAddress = sectionRva,
+                            Size = sectionSize,
+                            FileOffset = sectionDataOffset
+                        };
+                        sectionOffset += SectionHeaderSize;
+                    }
+
+                    uint cliHeaderOffset = RvaToOffset(sections, cliHeaderRva);
+
+                    // CLI section not found
+                    if (cliHeaderOffset == 0)
+                        return string.Empty;
+
+                    // The CLI header is specified in section II.25.3.3.
+                    // It contains all of the runtime-specific data entries and other information.
+                    // From the CLI header we need to get the RVA of the metadata root,
+                    // which is located at offset 8.
+
+                    sr.BaseStream.Position = cliHeaderOffset + 8;
+                    var metadataRva = sr.ReadUInt32();
+
+                    var metadataOffset = RvaToOffset(sections, metadataRva);
+                    if (metadataOffset == 0)
+                        return string.Empty;
+
+                    // The metadata root is specified in section II.24.2.1
+                    // The first 4 bytes contain a signature.
+                    // The version string is at offset 12.
+
+                    sr.BaseStream.Position = metadataOffset;
+                    if (!ReadBytes(sr, 0x42, 0x53, 0x4a, 0x42)) // Metadata root signature
+                        return string.Empty;
+
+                    // Read the version string length
+                    sr.BaseStream.Position = metadataOffset + 12;
+                    var length = sr.ReadInt32();
+                    if (length > 255 || length <= 0 || sr.BaseStream.Position + length >= sr.BaseStream.Length)
+                        return string.Empty;
+
+                    // Read the version string
+                    var v = Encoding.UTF8.GetString(sr.ReadBytes(length));
```

Comment:
The string thus generated has extra \0 bytes at the end. This parses OK into a Version, but does not compare to a version string. These extra characters should be trimmed. Alternatively, you could return below:
```C#
    return version.ToString();
```
The aforementioned TestGetImageRuntimeVersion can be used to verify that the version string is good.